src/lib/field.ts

```diff
@@ -908,34 +909,43 @@ class Field {
    * Returns an array of {@link Bool} elements representing [little endian binary representation](https://en.wikipedia.org/wiki/Endianness) of this {@link Field} element.
    *
    * If you use the optional `length` argument, proves that the field element fits in `length` bits.
-   * The `length` has to be between 0 and 255 and the method throws if it isn't.
+   * The `length` has to be between 0 and 254 and the method throws if it isn't.
    *
    * **Warning**: The cost of this operation in a zk proof depends on the `length` you specify,
-   * which by default is 255 bits. Prefer to pass a smaller `length` if possible.
+   * which by default is 254 bits. Prefer to pass a smaller `length` if possible.
    *
    * @param length - the number of bits to fit the element. If the element does not fit in `length` bits, the functions throws an error.
    *
    * @return An array of {@link Bool} element representing little endian binary representation of this {@link Field}.
    */
-  toBits(length?: number) {
-    if (length !== undefined) checkBitLength('Field.toBits()', length);
+  toBits(length: number = 254) {
+    checkBitLength('Field.toBits()', length, 254);
     if (this.isConstant()) {
       let bits = Fp.toBits(this.toBigInt());
-      if (length !== undefined) {
-        if (bits.slice(length).some((bit) => bit))
-          throw Error(`Field.toBits(): ${this} does not fit in ${length} bits`);
-        return bits.slice(0, length).map((b) => new Bool(b));
-      }
-      return bits.map((b) => new Bool(b));
+      if (bits.slice(length).some((bit) => bit))
+        throw Error(`Field.toBits(): ${this} does not fit in ${length} bits`);
+      return bits.slice(0, length).map((b) => new Bool(b));
     }
-    let [, ...bits] = Snarky.field.toBits(length ?? Fp.sizeInBits, this.value);
-    return bits.map((b) => new Bool(b));
+    let bits = Provable.witness(Provable.Array(Bool, length), () => {
+      let f = this.toBigInt();
+      return Array.from(
+        { length },
+        (_, k) => new Bool(!!((f >> BigInt(k)) & 0x1n))
+      );
+    });
+    Field.fromBits(bits).assertEquals(
+      this,
+      `Field.toBits(): ${Provable.asProver(() => {
```

Comment:
Is `Provable.asProver()` the correct way to log this?

Reply:
what lol no, this will log `undefined` because `asProver()` doesn't return anything.
I would leave out the value,
```suggestion
      `Field.toBits(): Input does not fit in ${length} bits`
```